### PR TITLE
mgr/client: validate connection before sending

### PR DESCRIPTION
**Pull request by NitzanMordhai**
During the connection's lifecycle, it can be reset and set to nullptr by other functions.
We need to verify that the connection is still available before attempting to send data.

Fixes: https://tracker.ceph.com/issues/62162

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/crimson/mgr/client.cc [hunk 1]
```diff
@@ -165,6 +165,10 @@ void Client::report()
     }
     return with_stats.get_stats(
     ).then([this](auto &&pg_stats) {
+      if (!conn) {
+        logger().warn("report: no conn available; report skipped");
```
[the remark this answers is not in the document]
Comment (NitzanMordhai) [reply src/crimson/mgr/client.cc]: @rzarzynski sure, will change it to before sending stats

**Review by rzarzynski — APPROVED** [review]

**Review by cyx1231st — APPROVED** [review]

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by cyx1231st** [comment]
jenkins test api

**Comment by NitzanMordhai** [comment]
ran with revert 752b4c6110590cdd85b04e6a01e6f1affcdee97c:
https://pulpito.ceph.com/nmordech-2024-07-16_08:34:36-crimson-rados-wip-nitzan-crimson-mgr-client-lost-connection-distro-default-smithi/
All pass

**Comment by NitzanMordhai** [comment]
Adding the revert commit to that PR as well

**Review by Matan-B — APPROVED** [review]

**Comment by Matan-B** [comment]
https://shaman.ceph.com/builds/ceph/wip-matanb-crimson-testing-july-25/

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-07-25_13:29:28-crimson-rados-wip-matanb-crimson-testing-july-25-distro-crimson-smithi/

7817766 - https://tracker.ceph.com/issues/67219

**Pull request merged**